app.js

```diff
@@ -56,6 +59,79 @@ env.addGlobal(
   )
 );
 
+let uri = "mongodb://localhost:27017";
+if (process.env.USE_PROD) {
+  const user = process.env.MONGO_USER;
+  const password = process.env.MONGO_PASSWORD;
+  const clusterUri = process.env.MONGO_URI;
+  uri = `mongodb+srv://${user}:${password}@${clusterUri}/test?retryWrites=true`;
+} 
+
+app.route("/allSongs").get(function(req, res) {
+  const client = new MongoClient(uri, { useNewUrlParser: true });
+
+  client.connect(err => {
+    if (err) {
+      throw err;
```

Comment:
Within an asynchronous callback function like that of `connect` or `findOne`, throwing an error will just kill the entire server process instead of aborting the current request.

The least error-prone solution would be using the [MongoDB `Promise` APIs](http://mongodb.github.io/node-mongodb-native/3.2/reference/ecmascriptnext/) with `async`/`await` instead of asynchronous callback functions, and also upgrading to Express 5.0.0-alpha7, which handles `Promise` rejections by converting them to 500 errors.